src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -491,9 +491,9 @@ public ParameterSymbol SynthesizedParameter(TypeSymbol type, string name, Method
             return SynthesizedParameterSymbol.Create(container, type, ordinal, RefKind.None, name);
         }
 
-        public BoundBinaryOperator Binary(BinaryOperatorKind kind, TypeSymbol type, BoundExpression left, BoundExpression right)
+        public BoundBinaryOperator Binary(BinaryOperatorKind kind, TypeSymbol type, BoundExpression left, BoundExpression right, MethodSymbol methodSymbolOpt = null)
```

Comment:
>MethodSymbol methodSymbolOpt = null [](start = 129, length = 35)

I think that the initial intent behind the factory was to provide an easy way to build nodes in lowered form, i.e. that don't require additional lowering. I am not sure there are other helpers that don't follow that, but it feels like this change would conflict with the intent and would make it easy for developers to build trees that won't be properly handled by emit layer.

Reply:
Good catch. This is a left-over from when I was double-lowering some nodes. This is no longer needed. Thanks

---
In reply to: [170486285](https://github.com/dotnet/roslyn/pull/23356#discussion_r170486285) [](ancestors = 170486285)